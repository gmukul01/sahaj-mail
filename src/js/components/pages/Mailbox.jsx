import React from 'react';

const Mailbox = ({ userName }) => {
	return (
		<div className="mailbox">
			<div className="sidebar">sidebar</div>
			<div className="header">header</div>
			<div className="content">content</div>
		</div>
	);
};

export default React.memo(Mailbox);
