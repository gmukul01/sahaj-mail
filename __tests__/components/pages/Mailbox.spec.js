import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';

import Mailbox from 'components/pages/Mailbox';

const mockStore = configureMockStore();

describe('Mailbox Component', () => {
	const initialState = {
			user: { name: 'dummy name', email: 'dummy@email.com' },
			inbox: {
				pageNumber: 1,
				totalEmails: 2,
				totalUnread: 2,
				emailsPerPage: 20,
				emails: [
					{
						id: 1,
						from: {
							name: 'dummyFrom1',
							email: 'dummyFrom1@email.com'
						},
						isRead: true
					},
					{
						id: 2,
						from: {
							name: 'dummyFrom2',
							email: 'dummyFrom2@email.com'
						},
						isRead: false
					}
				]
			}
		},
		store = mockStore(initialState);

	it('should match snapshot', () => {
		expect(
			renderer
				.create(
					<Provider store={store}>
						<Mailbox />
					</Provider>
				)
				.toJSON()
		).toMatchSnapshot();
	});
});
