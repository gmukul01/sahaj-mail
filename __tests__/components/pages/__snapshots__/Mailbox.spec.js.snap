// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mailbox Component should match snapshot when path is / 1`] = `
<div
  className="mailbox"
>
  <section
    className="sidebar"
    onClick={[Function]}
  >
    <button
      className="sidebar-row logo"
    >
      IN +
    </button>
    <div
      className="sidebar-row "
    >
      <button>
        <i
          className="fas fa-th-large"
        />
        <p>
          Dashboard
        </p>
      </button>
      <i
        className="fas fa-angle-left sidebar-dropdown"
      />
    </div>
    <div
      className="sidebar-row "
    >
      <button>
        <i
          className="fas far fa-gem"
        />
        <p>
          Layout
        </p>
      </button>
      <i
        className="fas fa-angle-left sidebar-dropdown"
      />
    </div>
    <div
      className="sidebar-row "
    >
      <button>
        <i
          className="fas fa-chart-bar"
        />
        <p>
          Graphs
        </p>
      </button>
      <i
        className="fas fa-angle-left sidebar-dropdown"
      />
    </div>
    <div
      className="sidebar-row active"
    >
      <button>
        <i
          className="fas fa-envelope"
        />
        <p>
          Mailbox
        </p>
      </button>
      <i
        className="fas fa-angle-left sidebar-dropdown"
      />
    </div>
    <div
      className="sidebar-row "
    >
      <button>
        <i
          className="fas fa-chart-pie"
        />
        <p>
          Metrics
        </p>
      </button>
      <i
        className="fas fa-angle-left sidebar-dropdown"
      />
    </div>
    <div
      className="sidebar-row "
    >
      <button>
        <i
          className="fas fa-flask"
        />
        <p>
          Widgets
        </p>
      </button>
      <i
        className="fas fa-angle-left sidebar-dropdown"
      />
    </div>
    <div
      className="sidebar-row "
    >
      <button>
        <i
          className="fas far fa-sticky-note"
        />
        <p>
          Notes
        </p>
      </button>
      <i
        className="fas fa-angle-left sidebar-dropdown"
      />
    </div>
    <div
      className="sidebar-row "
    >
      <button>
        <i
          className="fas fas fa-tv"
        />
        <p>
          App Views
        </p>
      </button>
      <i
        className="fas fa-angle-left sidebar-dropdown"
      />
    </div>
    <div
      className="sidebar-row "
    >
      <button>
        <i
          className="fas far fa-copy"
        />
        <p>
          Random
        </p>
      </button>
      <i
        className="fas fa-angle-left sidebar-dropdown"
      />
    </div>
    <div
      className="sidebar-row "
    >
      <button>
        <i
          className="fas fas fa-globe-asia"
        />
        <p>
          Globe
        </p>
      </button>
      <i
        className="fas fa-angle-left sidebar-dropdown"
      />
    </div>
    <div
      className="sidebar-row "
    >
      <button>
        <i
          className="fas fa-flask"
        />
        <p>
          Widgets
        </p>
      </button>
      <i
        className="fas fa-angle-left sidebar-dropdown"
      />
    </div>
    <div
      className="sidebar-row "
    >
      <button>
        <i
          className="fas fas fa-laptop"
        />
        <p>
          Apps
        </p>
      </button>
      <i
        className="fas fa-angle-left sidebar-dropdown"
      />
    </div>
  </section>
  <section
    className="header"
  >
    <div
      className="header-searchbar"
    >
      <button
        onClick={[Function]}
      >
        <i
          className="fas fa-bars"
        />
      </button>
      <input
        onChange={[Function]}
        placeholder="Search for something..."
        type="text"
        value=""
      />
    </div>
    <div
      className="header-info"
    >
      <button
        className="badge-email"
      >
        <i
          className="fas fa-envelope"
        />
        <span
          className="badge"
        >
          2
        </span>
      </button>
      <button
        className="badge-notification"
      >
        <i
          className="fas fa-bell"
        />
        <span
          className="badge"
        >
          4
        </span>
      </button>
      <button
        className="logout"
        onClick={[Function]}
      >
        <i
          className="fas fa-sign-out-alt"
        />
        Logout
      </button>
    </div>
  </section>
  <section
    className="menubar"
  >
    <button
      className="button-solid menubar-compose"
      onClick={[Function]}
    >
      Compose Mail
    </button>
    <div
      className="modal"
    >
      <div
        className="modal-content"
      >
        <div
          className="modal-header"
        >
          <h3
            className="title"
          >
            New Message
          </h3>
          <button
            className="modal-close"
            onClick={[Function]}
          >
            Ã—
          </button>
        </div>
        <div
          className="modal-body"
        >
          <form
            className="compose-email"
            onSubmit={[Function]}
          >
            <label>
              To
            </label>
            <input
              onChange={[Function]}
              placeholder="Enter recipient email id"
              type="text"
              value=""
            />
            <label>
              Subject
            </label>
            <input
              onChange={[Function]}
              placeholder="Enter subject"
              type="text"
              value=""
            />
            <label>
              Body
            </label>
            <textarea
              onChange={[Function]}
              placeholder="Enter message"
              value=""
            />
            <button
              className="button-solid"
              type="submit"
            >
              Send
            </button>
          </form>
          
        </div>
      </div>
    </div>
    <nav
      className="menubar-folders"
    >
      <h5>
        FOLDERS
      </h5>
      <ul>
        <li>
          <a
            className=""
            href="/"
            onClick={[Function]}
          >
            <span>
              <i
                className="fas fa-inbox"
              />
              Inbox
            </span>
            <button
              className="menubar-folders-inbox chips-btn"
            >
              2
            </button>
          </a>
        </li>
        <li>
          <a
            className="active"
            href="/sent"
            onClick={[Function]}
          >
            <span>
              <i
                className="fas fa-envelope"
              />
              Send Email
            </span>
          </a>
        </li>
        <li>
          <a
            className=""
            href="/"
            onClick={[Function]}
          >
            <span>
              <i
                className="fas fa-certificate"
              />
              Important
            </span>
          </a>
        </li>
        <li>
          <a
            className=""
            href="/"
            onClick={[Function]}
          >
            <span>
              <i
                className="fas fa-file-alt"
              />
              Drafts
            </span>
            <button
              className="menubar-folders-drafts chips-btn"
            >
              2
            </button>
          </a>
        </li>
        <li>
          <a
            className=""
            href="/"
            onClick={[Function]}
          >
            <span>
              <i
                className="fas fa-trash-alt"
              />
              Trash
            </span>
          </a>
        </li>
      </ul>
    </nav>
    <div
      className="menubar-categories"
    >
      <h5>
        CATEGORIES
      </h5>
      <ul>
        <li
          className="menubar-categories-work"
        >
          <i
            className="fas fa-circle"
          />
          Work
        </li>
        <li
          className="menubar-categories-documents"
        >
          <i
            className="fas fa-circle"
          />
          Documents
        </li>
        <li
          className="menubar-categories-social"
        >
          <i
            className="fas fa-circle"
          />
          Social
        </li>
        <li
          className="menubar-categories-advertising"
        >
          <i
            className="fas fa-circle"
          />
          Advertising
        </li>
        <li
          className="menubar-categories-clients"
        >
          <i
            className="fas fa-circle"
          />
          Clients
        </li>
      </ul>
    </div>
    <div
      className="menubar-labels"
    >
      <h5>
        LABELS
      </h5>
      <div>
        <button
          className="button-outlined"
        >
          <i
            className="fas fa-tag"
          />
          Film
        </button>
        <button
          className="button-outlined"
        >
          <i
            className="fas fa-tag"
          />
          Photography
        </button>
        <button
          className="button-outlined"
        >
          <i
            className="fas fa-tag"
          />
          Music
        </button>
        <button
          className="button-outlined"
        >
          <i
            className="fas fa-tag"
          />
          Holidays
        </button>
        <button
          className="button-outlined"
        >
          <i
            className="fas fa-tag"
          />
          Children
        </button>
        <button
          className="button-outlined"
        >
          <i
            className="fas fa-tag"
          />
          Home
        </button>
        <button
          className="button-outlined"
        >
          <i
            className="fas fa-tag"
          />
          Work
        </button>
        <button
          className="button-outlined"
        >
          <i
            className="fas fa-tag"
          />
          Family
        </button>
      </div>
    </div>
  </section>
  <section
    className="content mailBoxInfo"
  >
    <div
      className="mailBoxInfo-topbar"
    >
      <div
        className="title"
      >
        Sent ()
      </div>
    </div>
    <div
      className="mailBoxInfo-actionbar"
    >
      <div
        className="actions"
      >
        <button
          className="button-outlined actions-refresh"
          onClick={[Function]}
        >
          <i
            className="fas fa-sync-alt"
          />
          Refresh
        </button>
        <button
          className="button-outlined"
          onClick={[Function]}
        >
          <i
            className="far fa-eye"
          />
        </button>
        <button
          className="button-outlined"
        >
          <i
            className="fas fa-exclamation"
          />
        </button>
        <button
          className="button-outlined"
          onClick={[Function]}
        >
          <i
            className="far fa-trash-alt"
          />
        </button>
      </div>
      <div>
        <button
          className="button-outlined"
        >
          <i
            className="fas fa-arrow-left"
          />
        </button>
        <button
          className="button-outlined"
        >
          <i
            className="fas fa-arrow-right"
          />
        </button>
      </div>
    </div>
    <div
      className="email-list"
    />
  </section>
</div>
`;

exports[`Mailbox Component should match snapshot when path is /sent 1`] = `
<div
  className="mailbox"
>
  <section
    className="sidebar"
    onClick={[Function]}
  >
    <button
      className="sidebar-row logo"
    >
      IN +
    </button>
    <div
      className="sidebar-row "
    >
      <button>
        <i
          className="fas fa-th-large"
        />
        <p>
          Dashboard
        </p>
      </button>
      <i
        className="fas fa-angle-left sidebar-dropdown"
      />
    </div>
    <div
      className="sidebar-row "
    >
      <button>
        <i
          className="fas far fa-gem"
        />
        <p>
          Layout
        </p>
      </button>
      <i
        className="fas fa-angle-left sidebar-dropdown"
      />
    </div>
    <div
      className="sidebar-row "
    >
      <button>
        <i
          className="fas fa-chart-bar"
        />
        <p>
          Graphs
        </p>
      </button>
      <i
        className="fas fa-angle-left sidebar-dropdown"
      />
    </div>
    <div
      className="sidebar-row active"
    >
      <button>
        <i
          className="fas fa-envelope"
        />
        <p>
          Mailbox
        </p>
      </button>
      <i
        className="fas fa-angle-left sidebar-dropdown"
      />
    </div>
    <div
      className="sidebar-row "
    >
      <button>
        <i
          className="fas fa-chart-pie"
        />
        <p>
          Metrics
        </p>
      </button>
      <i
        className="fas fa-angle-left sidebar-dropdown"
      />
    </div>
    <div
      className="sidebar-row "
    >
      <button>
        <i
          className="fas fa-flask"
        />
        <p>
          Widgets
        </p>
      </button>
      <i
        className="fas fa-angle-left sidebar-dropdown"
      />
    </div>
    <div
      className="sidebar-row "
    >
      <button>
        <i
          className="fas far fa-sticky-note"
        />
        <p>
          Notes
        </p>
      </button>
      <i
        className="fas fa-angle-left sidebar-dropdown"
      />
    </div>
    <div
      className="sidebar-row "
    >
      <button>
        <i
          className="fas fas fa-tv"
        />
        <p>
          App Views
        </p>
      </button>
      <i
        className="fas fa-angle-left sidebar-dropdown"
      />
    </div>
    <div
      className="sidebar-row "
    >
      <button>
        <i
          className="fas far fa-copy"
        />
        <p>
          Random
        </p>
      </button>
      <i
        className="fas fa-angle-left sidebar-dropdown"
      />
    </div>
    <div
      className="sidebar-row "
    >
      <button>
        <i
          className="fas fas fa-globe-asia"
        />
        <p>
          Globe
        </p>
      </button>
      <i
        className="fas fa-angle-left sidebar-dropdown"
      />
    </div>
    <div
      className="sidebar-row "
    >
      <button>
        <i
          className="fas fa-flask"
        />
        <p>
          Widgets
        </p>
      </button>
      <i
        className="fas fa-angle-left sidebar-dropdown"
      />
    </div>
    <div
      className="sidebar-row "
    >
      <button>
        <i
          className="fas fas fa-laptop"
        />
        <p>
          Apps
        </p>
      </button>
      <i
        className="fas fa-angle-left sidebar-dropdown"
      />
    </div>
  </section>
  <section
    className="header"
  >
    <div
      className="header-searchbar"
    >
      <button
        onClick={[Function]}
      >
        <i
          className="fas fa-bars"
        />
      </button>
      <input
        onChange={[Function]}
        placeholder="Search for something..."
        type="text"
        value=""
      />
    </div>
    <div
      className="header-info"
    >
      <button
        className="badge-email"
      >
        <i
          className="fas fa-envelope"
        />
        <span
          className="badge"
        >
          2
        </span>
      </button>
      <button
        className="badge-notification"
      >
        <i
          className="fas fa-bell"
        />
        <span
          className="badge"
        >
          4
        </span>
      </button>
      <button
        className="logout"
        onClick={[Function]}
      >
        <i
          className="fas fa-sign-out-alt"
        />
        Logout
      </button>
    </div>
  </section>
  <section
    className="menubar"
  >
    <button
      className="button-solid menubar-compose"
      onClick={[Function]}
    >
      Compose Mail
    </button>
    <div
      className="modal"
    >
      <div
        className="modal-content"
      >
        <div
          className="modal-header"
        >
          <h3
            className="title"
          >
            New Message
          </h3>
          <button
            className="modal-close"
            onClick={[Function]}
          >
            Ã—
          </button>
        </div>
        <div
          className="modal-body"
        >
          <form
            className="compose-email"
            onSubmit={[Function]}
          >
            <label>
              To
            </label>
            <input
              onChange={[Function]}
              placeholder="Enter recipient email id"
              type="text"
              value=""
            />
            <label>
              Subject
            </label>
            <input
              onChange={[Function]}
              placeholder="Enter subject"
              type="text"
              value=""
            />
            <label>
              Body
            </label>
            <textarea
              onChange={[Function]}
              placeholder="Enter message"
              value=""
            />
            <button
              className="button-solid"
              type="submit"
            >
              Send
            </button>
          </form>
          
        </div>
      </div>
    </div>
    <nav
      className="menubar-folders"
    >
      <h5>
        FOLDERS
      </h5>
      <ul>
        <li>
          <a
            className=""
            href="/"
            onClick={[Function]}
          >
            <span>
              <i
                className="fas fa-inbox"
              />
              Inbox
            </span>
            <button
              className="menubar-folders-inbox chips-btn"
            >
              2
            </button>
          </a>
        </li>
        <li>
          <a
            className="active"
            href="/sent"
            onClick={[Function]}
          >
            <span>
              <i
                className="fas fa-envelope"
              />
              Send Email
            </span>
          </a>
        </li>
        <li>
          <a
            className=""
            href="/"
            onClick={[Function]}
          >
            <span>
              <i
                className="fas fa-certificate"
              />
              Important
            </span>
          </a>
        </li>
        <li>
          <a
            className=""
            href="/"
            onClick={[Function]}
          >
            <span>
              <i
                className="fas fa-file-alt"
              />
              Drafts
            </span>
            <button
              className="menubar-folders-drafts chips-btn"
            >
              2
            </button>
          </a>
        </li>
        <li>
          <a
            className=""
            href="/"
            onClick={[Function]}
          >
            <span>
              <i
                className="fas fa-trash-alt"
              />
              Trash
            </span>
          </a>
        </li>
      </ul>
    </nav>
    <div
      className="menubar-categories"
    >
      <h5>
        CATEGORIES
      </h5>
      <ul>
        <li
          className="menubar-categories-work"
        >
          <i
            className="fas fa-circle"
          />
          Work
        </li>
        <li
          className="menubar-categories-documents"
        >
          <i
            className="fas fa-circle"
          />
          Documents
        </li>
        <li
          className="menubar-categories-social"
        >
          <i
            className="fas fa-circle"
          />
          Social
        </li>
        <li
          className="menubar-categories-advertising"
        >
          <i
            className="fas fa-circle"
          />
          Advertising
        </li>
        <li
          className="menubar-categories-clients"
        >
          <i
            className="fas fa-circle"
          />
          Clients
        </li>
      </ul>
    </div>
    <div
      className="menubar-labels"
    >
      <h5>
        LABELS
      </h5>
      <div>
        <button
          className="button-outlined"
        >
          <i
            className="fas fa-tag"
          />
          Film
        </button>
        <button
          className="button-outlined"
        >
          <i
            className="fas fa-tag"
          />
          Photography
        </button>
        <button
          className="button-outlined"
        >
          <i
            className="fas fa-tag"
          />
          Music
        </button>
        <button
          className="button-outlined"
        >
          <i
            className="fas fa-tag"
          />
          Holidays
        </button>
        <button
          className="button-outlined"
        >
          <i
            className="fas fa-tag"
          />
          Children
        </button>
        <button
          className="button-outlined"
        >
          <i
            className="fas fa-tag"
          />
          Home
        </button>
        <button
          className="button-outlined"
        >
          <i
            className="fas fa-tag"
          />
          Work
        </button>
        <button
          className="button-outlined"
        >
          <i
            className="fas fa-tag"
          />
          Family
        </button>
      </div>
    </div>
  </section>
  <section
    className="content mailBoxInfo"
  >
    <div
      className="mailBoxInfo-topbar"
    >
      <div
        className="title"
      >
        Sent ()
      </div>
    </div>
    <div
      className="mailBoxInfo-actionbar"
    >
      <div
        className="actions"
      >
        <button
          className="button-outlined actions-refresh"
          onClick={[Function]}
        >
          <i
            className="fas fa-sync-alt"
          />
          Refresh
        </button>
        <button
          className="button-outlined"
          onClick={[Function]}
        >
          <i
            className="far fa-eye"
          />
        </button>
        <button
          className="button-outlined"
        >
          <i
            className="fas fa-exclamation"
          />
        </button>
        <button
          className="button-outlined"
          onClick={[Function]}
        >
          <i
            className="far fa-trash-alt"
          />
        </button>
      </div>
      <div>
        <button
          className="button-outlined"
        >
          <i
            className="fas fa-arrow-left"
          />
        </button>
        <button
          className="button-outlined"
        >
          <i
            className="fas fa-arrow-right"
          />
        </button>
      </div>
    </div>
    <div
      className="email-list"
    />
  </section>
</div>
`;
